### PR TITLE
SZ compression

### PR DESCRIPTION
**Pull request by jychoi-hpc**
I finished adding SZ compression as an operator in Adios2. This includes
- SZ cmake find module
- SZ operator
- An example

### CODE REVIEW

File: cmake/FindSZ.cmake [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    set(SZ_INCLUDE_OPTS HINTS ${SZ_ROOT}/include NO_DEFAULT_PATHS)
+    set(SZ_LIBRARY_OPTS
+      HINTS ${SZ_ROOT}/lib ${SZ_ROOT}/lib64
+      NO_DEFAULT_PATHS
+    )
+  endif()
+
+  find_path(SZ_INCLUDE_DIR sz.h ${SZ_INCLUDE_OPTS})
+  find_library(SZ_LIBRARY NAMES SZ ${SZ_LIBRARY_OPTS})
+  find_library(ZLIB_LIBRARY NAMES zlib ${SZ_LIBRARY_OPTS})
```
Comment (chuckatkins) [inline cmake/FindSZ.cmake]: Is zlib also a dependecy?  If so this should be dropped and instead added via find_package(zlib)

**Comment by jychoi-hpc** [comment]
SZ distribute their library together with ZLIB. I will check if we can use different version of ZLIB installed in a system.

Thanks,
Jong

-- 
Jong Youl Choi
Scientific Data Group
Computer Science and Math Division
Oak Ridge National Laboratory
Homepage: http://www.ornl.gov/~jyc/

**Comment by williamfgc** [comment]
@jychoi-hpc thanks for the great work! To pass the travis-ci check we use the clang-format (v3.8 recommended on Mac and llvm style). It's just a binary that need to be applied to your files, see our document: https://github.com/ornladios/ADIOS2/wiki/Clang-Format-Text-Style Looking forward to SZ support. Thanks!

**Comment by jychoi-hpc** [comment]
How can I get this "clang-format" installed on Mac? Can I install with macport or homebrew?

File: cmake/FindSZ.cmake (hunk 1 above)
Comment (jychoi-hpc) [reply cmake/FindSZ.cmake]: After a bit of investigation, it looks like SZ uses their own zlib library and include in their distribution along with SZ. Then, if this is the case, we don't need to use another "find_package(zlib)", right?

**Comment by eisenhauer** [comment]
http://brewformulas.org/ClangFormat

**Pull request merged**